tests/integrations/test_zk_evm.py

```diff
@@ -107,3 +107,20 @@ async def test_erc20(
             assert balances_after[2] - balances_before[2] == 0
             assert balances_after[3] - balances_before[3] == 0x5
         kakarot.state = state
+
+    @pytest.mark.IntegrationTestContract
+    async def test_integration_contract(
+        self, kakarot: StarknetContract, deploy_solidity_contract: Callable, request
+    ):
+        state = kakarot.state.copy()
+        caller_addresses = list(range(4))
+        integration_contract, tx = await deploy_solidity_contract(
+            "IntegrationTestContract", caller_address=caller_addresses[1]
+        )
+
+        with traceit.context(request.node.own_markers[0].name):
+
+            evm_contract_address = await integration_contract.opcodeAddress()
+            assert tx.result.evm_contract_address == int(evm_contract_address, 0)
```

Comment:
why do we have to cast `evm_contract_address`? also surprised by the `, 0)`

Reply:
So calling `opcodeAddress` gives us `evm_contract_address="0xAbDe100763b3fb83ae8C5aFE6c635eEBA23E276B"`. I had to cast it to `int(foo, 0)` as it's a string with prefix 0x in order to assert equality with the resulting evm contract address from `deploy_call_info`.

Reply:
Can't you do the opposite and cast tx.result.evm_contract_address in hex() ?

Reply:
@Eikix Ah yes that's true, base 0 interprets it exactly as a code literal but you're right it's better to be explicit! Indeed it's good to change this to `int(foo, 16)` instead 🤙

@matthieuauger Good idea, also works with hex but had to format it to uppercase like so `assert f"0x{..result.evm_contract_address:X} == evm_contract_address`. Perhaps using `int` is more readable.. should we use that instead?

Reply:
i guess it'd make sense to compare hexes as hexes are used to represent addresses, but I wonder what equality operator we'd be using -> we'd be asserting over string equality instead of integer equality?

Reply:
Yeah I agree, perhaps casting them both to hex and assert equality like this `hex(..result.evm_contract_address) == hex(int(evm_contract_address), 16)`

Reply:
what do you think is the problem with comparing both hexes?
to be fair, everything in the context of starknet is an int or at least a felt:)!
comparing both data as int lgtm!!!
i'll approve your PR with int() comparison, double casting (int and then hex to not break) feels like overdoing it for me

Reply:
int is easier in our context because python can handle BigInt natively